lib/filter-stream.js

```diff
@@ -190,6 +191,18 @@ function FilterStream( blockMap, options ) {
 
   },
 
+  _flush( done ) {
+
+    var buffer = Buffer.concat( this._buffers )
+    var block = Buffer.alloc( this.blockSize )
+
+    buffer.copy( block )
```

Comment:
Since `_transform()` will always run before a `_flush()`, and the (fixed) loop in the transform will emit block sized chunks, the leftover buffer will (theoretically) always be less than the block size.